inbox/mailsync/backends/imap/monitor.py

```diff
@@ -185,3 +186,14 @@ def sync(self):
                 account = db_session.query(Account).get(self.account_id)
                 account.mark_invalid()
                 account.update_sync_error(exc)
+
+    def stop(self) -> None:
+        from inbox.mailsync.backends.gmail import GmailSyncMonitor
+
+        if self.delete_handler:
+            self.delete_handler.kill()
+        kill_all(self.folder_monitors, block=False)
+        if isinstance(self, GmailSyncMonitor):
+            kill_all(self.label_rename_handlers.values(), block=False)
+        self.sync_greenlet.kill(block=False)
+        self.join()
```

Comment:
This is better contained in a function since you need to stop the whole "tree" of greenlets.